src/map/battle.c

```diff
@@ -7180,6 +7180,9 @@ static const struct battle_data {
 	{ "feature.roulette",                   &battle_config.feature_roulette,                1,      0,      1,              },
 	{ "show_monster_hp_bar",                &battle_config.show_monster_hp_bar,             1,      0,      1,              },
 	{ "fix_warp_hit_delay_abuse",           &battle_config.fix_warp_hit_delay_abuse,        0,      0,      1,              },
+	{ "costume_refine_def",                 &battle_config.costume_refine_def,              0,      0,      1,              },
+	{ "shadow_refine_def",                  &battle_config.shadow_refine_def,               0,      0,      1,              },
+	{ "shadow_refine_atk",                  &battle_config.shadow_refine_atk,               0,      0,      1,              },
```

Comment:
in the battle/player.conf, you said default to `yes`, but over here, it default to `0`
it should be
`{ "shadow_refine_atk",                  &battle_config.shadow_refine_atk,               1,      0,      1,              },`